raster-foundry/raster-foundry
# Add lab node histograms and a default color ramp

## Overview

Add histograms to lab nodes

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo
![image](https://user-images.githubusercontent.com/4392704/30074347-3e8a1c42-923f-11e7-9852-3db4db5dccc0.png)
![image](https://user-images.githubusercontent.com/4392704/30074375-570c9ea2-923f-11e7-88ce-e17912091ff0.png)
![image](https://user-images.githubusercontent.com/4392704/30074395-66584f32-923f-11e7-8255-83801ea51a42.png)
![image](https://user-images.githubusercontent.com/4392704/30074412-751e717c-923f-11e7-9f68-d62dcc6c1c51.png)
![image](https://user-images.githubusercontent.com/4392704/30074785-be18c476-9240-11e7-84b3-f15e218e11d1.png)


### Notes
Builds on top of #2475 
Waiting on #2467

## Testing Instructions

 * Verify that the warning message shows if you open the histogram before applying changes
* Select projects / inputs for the tool run and hit apply
* View a node (without opening the histogram yet) and verify that it has a default color ramp
* open a histogram and hit apply, then verify that it uses the settings in the histogram
* Change the values, on the histogram (min, max checkboxes and breakpoints) and verify that they work as expected. Comparing two nodes with the same project / band and different color ramps may be helpful.

Closes #2387 
Closes #2389
